internal/utils/utils.go

```diff
@@ -171,3 +174,23 @@ func InPool(id string, list []civogo.KubernetesClusterPoolConfig) bool {
 	}
 	return false
 }
+
+// FunctionWithError is a type that defines a function returning an error.
+type FunctionWithError func() error
+
+// RetryUntilSuccessOrTimeout calls the provided function repeatedly until it returns no error or the timeout has passed.
+func RetryUntilSuccessOrTimeout(fn FunctionWithError, interval time.Duration, timeout time.Duration) error {
```

Comment:
Can we have a dedicate package called `retry` or `ctrl`(control) to have such functions there?

Reply:
Dosen't make scence to me as i dont think we will have any more such related functions and if we do we can ofc put it in utils package. Looks like an overhead to me to make a new package just for this